test/test.js

```diff
@@ -69,33 +69,30 @@ describe('onclickoutside hoc', function() {
 
 
   it('should call the specified handler when clicking the document', function() {
-    var Component = React.createClass({
-      getInitialState: function() {
-        return {
+    class Component extends React.Component {
```

Comment:
Still thinking about your comment, and stateless functions won't be re-bound here, because they do not inherit from React.Component. I'm not sure if it makes sense to support instance method on stateless functions, but if so, the test should be reversed to check if createClass was used. The (very ugly) way to do that appears to be:

```javascript
var ReactClassComponentPrototype = (Object.getPrototypeOf(Object.getPrototypeOf(new (React.createClass({ render: function () {} }))())))
```

Source: https://discuss.reactjs.org/t/how-to-determine-if-js-object-is-react-component/2825/5